### ISSUE TITLE
[BUG] When calling `getAdvancedHistoryMessageList` in a group, it generates a large number of `get_users_info` requests

### ISSUE THREAD

**Issue by cipchk** [openimsdk/open-im-server#2831]
### OpenIM Server Version

3.8.1

### Operating System and CPU Architecture

Linux (AMD)

### Deployment Method

Docker Deployment

### Bug Description and Steps to Reproduce

Here is the request code:

```
await OpenIM.iMManager.messageManager.getAdvancedHistoryMessageList(
      conversationID: conversationInfo.conversationID,
      count: 20,
      startMsg: _isFirstLoad ? null : messageList.firstOrNull,
      lastMinSeq: _isFirstLoad ? null : lastMinSeq,
    );
```

Requesting 20 records generates more than 40 `get_users_info` requests, see PNG:

<img width="599" alt="image" src="https://github.com/user-attachments/assets/e31e4745-2874-46d6-9006-3053c003e408">

This is the specific log of one of the requests and responses, and there are over 40 similar ones. The only difference is that they all have different userIDs:

```
I/GoLog   (30089): ]	[flutter/Android]                                 	[network/http_client.go:96]                       	ApiRequest                                        	{"operationID": "1730706607309", "url": "https://xxx.com/api/user/get_users_info", "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJVc2VySUQiOiIxMDAwMCIsIlBsYXRmb3JtSUQiOjIsImV4cCI6MTczODA2MTgwMSwibmJmIjoxNzMwMjg1NTAxLCJpYXQiOjE3MzAyODU4MDF9.N6Qs5IFqiK8qAEOAFvxa96b3Sw8o9RBxyw89GRC-Jzs", "body": "{\"userIDs\":[\"10563\"]}"}
I/GoLog   (30089): ]	[flutter/Android]                                 	[network/http_client.go:131]                      	ApiResponse                                       	{"operationID": "1730706607309", "url": "https://xxx.com/api/user/get_users_info", "status": "200 OK", "body": "{\"errCode\":0,\"errMsg\":\"\",\"errDlt\":\"\",\"data\":{\"usersInfo\":[{\"userID\":\"10563\",\"nickname\":\"Натали\",\"faceURL\":\"\",\"ex\":\"\",\"createTime\":1725811767651,\"appMangerLevel\":0,\"globalRecvMsgOpt\":0}]}}"}
```

**Comment by cipchk** [comment]
@icey-yu 有任何进展吗？因为这个情况导致我们现在都不敢让所有用户都使用 3.8.1 版本。

**Comment by icey-yu** [comment]
We are investigating the issue.

**Comment by icey-yu** [comment]
In our tests, the SDK's interface calls are functioning normally. The logs you filtered may include entries from other calls. You can use the operationID to confirm if it’s from the same call.

**Comment by cipchk** [comment]
@icey-yu Yes, they all share the same operationid, and I can confirm that it gets generated when calling `getAdvancedHistoryMessageList`.

**Comment by icey-yu** [comment]
Is the conversation a group chat, and are most participants in the group non-friends?

**Comment by cipchk** [comment]
@icey-yu Yes, over 99% of them are not friends with each other.

**Comment by icey-yu** [comment]
We optimized this issue a few days ago. You can pull the latest code from the main branch of https://github.com/openimsdk/openim-sdk-core and try again.

**Comment by cipchk** [comment]
Could you provide a specific PR related to optimization?

**Comment by icey-yu** [comment]
https://github.com/openimsdk/openim-sdk-core/pull/765

**Comment by FGadvancer** [comment]
This issue has been fixed in the next 3.8.2 release.
